File: gateway/src/main/java/com/undancer/app/IndexController.java
package com.undancer.app;

import com.google.common.collect.Maps;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping(path = "/")
public class IndexController {

    @GetMapping
    public Object home() {
        Map<String, Object> resultObject = Maps.newHashMap();
        resultObject.put("username", "undancer");
        return resultObject;
    }

}

File: gateway/src/main/kotlin/com/undancer/cloud/gateway/IndexController.kt
package com.undancer.cloud.gateway

import com.google.common.collect.Maps
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = arrayOf("/"))
class IndexController {

    @GetMapping
    fun home(): Any {
        val resultObject = Maps.newHashMap<String, Any>()
        resultObject["username"] = "undancer"
        return resultObject
    }

}
